crates/ark/src/treesitter.rs

```diff
@@ -404,12 +410,37 @@ impl NodeTypeExt for Node<'_> {
     fn is_binary_operator(&self) -> bool {
         matches!(self.node_type(), NodeType::BinaryOperator(_))
     }
+
+    fn find_parent<Callback: Fn(Node) -> bool>(&self, callback: Callback) -> Option<Node> {
+        let mut node = *self;
+
+        while let Some(parent) = node.parent() {
+            if callback(parent) {
+                return Some(parent);
+            }
+            node = parent;
+        }
+
+        None
+    }
```

Comment:
Questionably efficient but it works for this use case

Reply:
hmm I _think_ we don't need this method? Instead we can use the `ancestors()` method I added with the `find()` iterator method?

Or at least implement in terms of these.

Reply:
Nice I forgot about that